core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1151,6 +1149,16 @@ class LogManager(logDirs: Seq[File],
       }
     }
   }
+
+  private def removeLogAndMetrics(logs: Pool[TopicPartition, Log], tp: TopicPartition): Option[Log] = {
+    val removedLog = logs.remove(tp)
+    if (removedLog != null) {
+      removedLog.removeLogMetrics()
+      Some(removedLog)
+    } else {
+      None
+    }
```

Comment:
nit: I think just returning `Option(removedLog)` should take care of Some/None part.